types/react-router/test/WithRouter.tsx

```diff
@@ -22,3 +22,31 @@ const WithRouterTestFunction = () => (
     <WithRouterComponentFunction username="John" />
 );
 const WithRouterTestClass = () => <WithRouterComponentClass username="John" />;
+
+// union props
+{
+    interface Book {
+        kind: 'book';
+        author: string;
+    }
+
+    interface Magazine {
+        kind: 'magazine';
+        issue: number;
+    }
+
+    type SomethingToRead = (Book | Magazine) & RouteComponentProps;
+
+    const Readable: React.SFC<SomethingToRead> = props => {
+        if (props.kind === 'magazine') {
```

Comment:
Might be personal preference but I think you can simplify the rendering code and get rid of the `if` statement by using ternary:

```
return <div>magazine #{props.kind === 'magazine' ? props.issue : props.author}</div>
```

Reply:
I want to test types not code style.